Frontend/tradingbot/src/Login.js

```diff
@@ -0,0 +1,106 @@
+import { useRef, useState, useEffect } from 'react';
+import axios from './API/axios';
+import { useNavigate } from 'react-router-dom'
+import { BounceLoader } from "react-spinners";
+
+const LOGIN_URL = '/performance';
+const HOME_URL = '/home';
+
+const Login = () => {
+
+    const navigate = useNavigate();
+
+    const userNameRef = useRef();
+    const errRef = useRef();
+
+    const [userName, setUserName] = useState('');
+    const [pwd, setPwd] = useState('');
+    const [errMsg, setErrMsg] = useState('');
+
+    const [isLoading, setIsLoading] = useState(false);
+
+    useEffect(() => {
+        userNameRef.current.focus();
+    }, [])
+
+    useEffect(() => {
+        setErrMsg('');
+    }, [userName, pwd])
+
+    const handleSubmit = async (e) => {
+        e.preventDefault();
+        setErrMsg('');
+        try{
+            setIsLoading(true);
+            const response = await axios.get('',
+                {
+                    auth: {
+                        username: userName,
+                        password: pwd
+                    }
+                }
+            );
+            localStorage.setItem("userName", userName);
+            localStorage.setItem("pwd", pwd);
+            localStorage.setItem("isLoggedIn", 'true');
+            setUserName('');
+            setPwd('');
+            setIsLoading(false);
+            navigate(HOME_URL);
+        }catch(err){
+            if(!err?.response) {
+                setErrMsg('No Server Response')
+            } else if(err.response?.status === 400) {
+                setErrMsg('Login Failed');
+            } else if(err.response?.status === 404){
+                setErrMsg('Account does not exist');
+            } else if(err.response?.status === 401 ){
+                setErrMsg('Incorrect login credentials');
+            } else {
+                setErrMsg('Login Failed');
+            }
```

Comment:
We should probably have a separate case for `401`, since it's an user (app user) error. We could just ask for credentials again.
In other scenarios we should display error and not allow for retries, since they will probably also fail. IDK about lack of response - we could go with either option for that.

Reply:
Also it would probably be helpful to show at least the status code in case of error, and maybe also the response (as string).

Reply:
Also we shouldn't need `404` case. If someone provides a wrong username, we should respond in the same way we do if the password is wrong (so `404`).